pkg/apis/go.mod

```
@@ -1,31 +1,31 @@
 module github.com/rancher/rancher/pkg/apis
 
-go 1.20
+go 1.21
```

As per recent [version changes](https://tip.golang.org/doc/toolchain#version):

```suggestion
go 1.21.0
```

---

Good point, thanks for catching. I'll update it to `1.21.6`.

---

As discussed offline with @pjbgf, we will leave this change to a follow-up specific PR, because it will require adding the Go `GOTOOLCHAIN` env var to Dockerfile.dapper, so then the Go toolchain can download and use the most up to date patch version of itself.